acceptance/tests/modules/install/long_path_name.rb

```diff
@@ -0,0 +1,36 @@
+test_name "puppet module install (long file path)"
+require 'puppet/acceptance/module_utils'
+extend Puppet::Acceptance::ModuleUtils
+
+confine :except, :platform => /centos-4|el-4/ # PUP-5226
+confine :except, :platform => /aix/ # PUP-5501
+
+module_author = 'ztr'
+module_name   = 'longpath'
```

Comment:
Note that your module needs to be of a specific tar format to expose the problem you're trying to expose.  I believe v7 format archives had the problem, but POSIX formats do not.